src/ui/control/geolocate_control.js

```diff
@@ -139,26 +373,65 @@ class GeolocateControl extends Evented {
+
+        this._geolocationWatchID = undefined;
+        this._geolocateButton.classList.remove('waiting');
+        this._geolocateButton.setAttribute('aria-pressed', false);
+
+        if (this.options.showUserLocation) {
+            this._updateMarker(null);
+        }
+    }
 }
 
 module.exports = GeolocateControl;
 
 /**
- * geolocate event.
+ * Fired on each Geolocation API position update which returned as success.
  *
  * @event geolocate
  * @memberof GeolocateControl
  * @instance
- * @property {Position} data The returned [Position](https://developer.mozilla.org/en-US/docs/Web/API/Position) object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition).
+ * @property {Position} data The returned [Position](https://developer.mozilla.org/en-US/docs/Web/API/Position) object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) or [Geolocation.watchPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition).
  *
  */
 
 /**
- * error event.
+ * Fired on each Geolocation API position update which returned as an error.
  *
  * @event error
  * @memberof GeolocateControl
  * @instance
- * @property {PositionError} data The returned [PositionError](https://developer.mozilla.org/en-US/docs/Web/API/PositionError) object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition).
+ * @property {PositionError} data The returned [PositionError](https://developer.mozilla.org/en-US/docs/Web/API/PositionError) object from the callback in [Geolocation.getCurrentPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) or [Geolocation.watchPosition()](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/watchPosition).
+ *
+ */
+
+/**
+ * Fired when the Geolocate Control is ready and able to be clicked.
+ *
+ * @event ready
+ * @memberof GeolocateControl
+ * @instance
+ *
+ */
+
+/**
+ * Fired when the Geolocate Control changes to the active lock state, which happens either upon first obtaining a successful Geolocation API position for the user (a geolocate event will follow), or the user clicks the geolocate button when in the background state which uses the last known position to recenter the map and enter active lock state (no geolocate event will follow unless the users's location changes).
+ *
+ * @event activeLock
+ * @memberof GeolocateControl
+ * @instance
+ *
+ */
+
+/**
+ * Fired when the Geolocate Control changes to the background state, which happens when a user changes the camera during an active position lock. This only applies when trackUserLocation is true. In the background state, the dot on the map will update with location updates but the camera will not.
+ *
+ * @event background
```

Comment:
What do you think about renaming these events to:

 - `activeLock` -> `trackuserlocationstart`
 - `background` -> `trackuserlocationend`

These names are consistent with existing event names (`start` and `end` suffixes) and more self-documenting (`activeLock` isn't a term of art AFIK). The all lowercase naming convention for events is unfortunate but consistent in our codebase so far.